src/lib/util/Config.ts

```diff
@@ -79,6 +90,31 @@ export class Config {
 		return this.configContent.engines.find(e => e.name === name);
 	}
 
+	public getOverriddenConfigPath(name: string): string {
+		const defaultValue = '';
+		const engineConfig = this.getEngineConfig(name);
+
+		if (!this.shouldUseDefaultConfig(name)) {
+			if (!engineConfig || !engineConfig.overriddenConfigPath) {
+				throw new SfdxError(`Please set "overriddenConfigPath" with config path to override for engine ${name} in ${CONFIG_FILE_PATH}. If you want to use default value, please set "useDefaultConfig" to true.`);
+			}
+			return engineConfig.overriddenConfigPath;
+		}
+		return defaultValue;
+	}
+
+	private shouldUseDefaultConfig(name: string): boolean {
+		const engineConfig = this.getEngineConfig(name);
+		const defaultValue = true;
+
+		if (engineConfig) {
+			if (engineConfig.useDefaultConfig != null) {
```

Comment:
Could this rely on truthiness and skip the null check for useDefaultConfig?

if (engineConfig) { return engineConfig.useDefaultConfig; }

Reply:
Probably not. We don't use this value for PMD and this would be non-existent at initialization. Also, since these are meant for users to modify, I feel like there's a chance the config isn't necessarily pristine.